### DISCUSSION TITLE
how to change github accounts

### DISCUSSION THREAD

**Discussion by zapdafish** [community/community#133562] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

My company bought me a copilot license. I signed up with github using my work address and added that acount to visual studio. Now there doesnt appear to be a way to logout to remove it so I can use the one my company provided. Everytime I click on remove then add account, it takes me to the github webpage and automatically adds that account

**Comment by SohanKafle** [comment]
Hi zapdafish,
<br>

It sounds like you're having trouble switching GitHub accounts in Visual Studio. Follow these steps to remove the old account and add the new one provided by your company:

<h1>Step-by-Step Guide to Change GitHub Accounts in Visual Studio</h1>
<h2>1. Remove GitHub Credentials:</h2>
First, you need to remove the saved GitHub credentials from your system.

<h3>Windows:</h3>
Open the Credential Manager (you can find it by searching in the Start menu).
Go to the Windows Credentials tab.
Find any entries related to github.com and remove them.
<h3>macOS:</h3>
Open Keychain Access.
Search for any items related to github.com and delete them.

<h2>2. Clear GitHub Tokens:</h2>
Remove any personal access tokens or OAuth tokens related to GitHub from Visual Studio.
Open Visual Studio.
Go to Tools > Options.
Navigate to Azure DevOps or Accounts (depending on the Visual Studio version).
Remove any entries related to the old GitHub account.

<h2>3. Sign Out of GitHub in Visual Studio:</h2>
Ensure you are signed out of GitHub in Visual Studio.
Go to File > Account Settings > All Accounts.
Sign out of the old GitHub account if it's listed there.

<h2>4. Sign Out in Web Browser:</h2>
Sign out of GitHub in your web browser to prevent automatic sign-ins.
Open your web browser.
Go to https://github.com/.
Sign out of the old GitHub account.

<h2>5. Sign In with New Account:</h2>
Add the new GitHub account in Visual Studio.
Open Visual Studio.
Go to File > Account Settings > Add an account.
When prompted, sign in with your work email address (the new GitHub account provided by your company).

<h2>6. Verify and Connect:</h2>
Verify that the new account is connected properly.
Go to View > GitHub.
Check if the new account is listed and you can see your repositories.

<h2>Additional Tips:</h2>
If you still face issues, restart Visual Studio and your system to ensure all changes take effect.
Double-check that no browser extensions or settings are automatically signing you in with the old account.
By following these steps, you should be able to successfully switch to the new GitHub account provided by your company. Let me know if you need any further assistance!
<br></br>


<b>Best regards,
Sohan Kafle</b>

**Comment by InboraStudio** [comment]
Step 1: Remove the Existing Account from Visual Studio
Sign Out from Visual Studio:

Open Visual Studio.
Go to File > Account Settings.
Click on Sign Out next to the account you want to remove.
Remove Account from Windows Credentials:

Open the Control Panel.
Go to User Accounts > Credential Manager.
Under Windows Credentials, find any credentials related to GitHub and remove them.
Step 2: Clear Browser Cookies and Cache
Before logging in with the new account, clear your browser cookies and cache to ensure the previous session does not auto-login.

Clear Cookies and Cache:
Open your browser.
Clear cookies and cache (usually found in Settings > Privacy and Security).
Step 3: Sign In with the New GitHub Account
Sign In to Visual Studio:
Open Visual Studio.
Go to File > Account Settings.
Click on Add an account.
This will open a browser window for GitHub login.
Enter the credentials for the new GitHub account provided by your company.
Step 4: Verify Copilot License
Verify Copilot Integration:
Ensure that your new GitHub account is correctly linked with the Copilot license.
You might need to go to the GitHub website to confirm that Copilot is activated for your new account.
Additional Tips
Use Incognito/Private Mode: If you still face issues with the auto-login, try using an incognito/private browsing window to sign in. This mode does not use stored cookies, which helps in avoiding automatic logins.

Browser Sessions: If you are logged in to the old account in any other browser session, log out from there as well.

Visual Studio Code (VS Code): If you are using Visual Studio Code (VS Code), the process is similar:

Go to File > Preferences > Settings.
Search for GitHub.
Click on Sign Out and then Sign In.

**Comment by queenofcorgis** [comment, marked as answer]
If you need to use multiple accounts on GitHub, you can sign in to your accounts and switch between them without always being required to reauthenticate. You can use the account switcher if you have a personal account and service accounts (sometimes called machine users) or if you need to switch between your personal account and managed user accounts in an enterprise that uses Enterprise Managed Users.

This [short video](https://www.youtube.com/shorts/Sznxc2d5UJk) can walk you through the process.
